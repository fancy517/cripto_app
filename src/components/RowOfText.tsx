import { Text } from "react-native-paper";
import { StyleSheet, View } from "react-native";
import React from "react";

type RowOfTextProps = {
  name: string;
  value: string | number;
};

const RowOfText = ({ name, value }: RowOfTextProps) => {
  return (
    <View style={style.divider}>
      <Text variant="bodyLarge">{name}</Text>
      <Text variant="bodyLarge">{value}</Text>
    </View>
  );
};

const style = StyleSheet.create({
  divider: {
    // flex: 1,
    flexDirection: "row",
    justifyContent: "space-between",
  },
});

export default RowOfText;
