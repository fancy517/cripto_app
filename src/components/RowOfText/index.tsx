import RowOfText from "./RowOfText";
export default RowOfText;
