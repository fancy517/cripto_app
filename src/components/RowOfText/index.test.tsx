import React from "react";
import renderer from "react-test-renderer";
import RowOfText from "./index";

const name = "Name";
const value = "Value";

describe("RowOfText", () => {
  it("matches snapshot", () => {
    const jsonElement = renderer
      .create(<RowOfText name={name} value={value} />)
      .toJSON();

    console.log(jsonElement);
    expect(jsonElement).toMatchSnapshot();
  });

  it("has 2 child", () => {
    const jsonElement = renderer
      .create(<RowOfText name={name} value={value} />)
      .toJSON();

    expect(jsonElement?.children.length).toBe(2);
  });
});
