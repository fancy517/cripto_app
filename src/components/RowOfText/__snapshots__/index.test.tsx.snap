// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowOfText matches snapshot 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          undefined,
        ],
      ]
    }
  >
    Name
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          undefined,
        ],
      ]
    }
  >
    Value
  </Text>
</View>
`;

exports[`RowOfText renders the component with name and value 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          undefined,
        ],
      ]
    }
  >
    Name
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          undefined,
        ],
      ]
    }
  >
    Value
  </Text>
</View>
`;
